Source/WebCore/page/IntelligenceTextEffectsSupport.cpp

```diff
@@ -109,5 +111,29 @@ std::optional<TextIndicatorData> textPreviewDataForRange(Document& document, con
     return textIndicator->data();
 }
 
+#if ENABLE(WRITING_TOOLS)
+void decorateWritingToolsTextReplacements(Document& document, const SimpleRange& scope, const CharacterRange& range)
+{
+    auto resolvedRange = resolveCharacterRange(scope, range);
+
+    auto& markers = document.markers();
+
+    markers.forEach(resolvedRange, { DocumentMarker::Type::WritingToolsTextSuggestion }, [&](auto& node, auto& marker) {
+        auto oldData = std::get<DocumentMarker::WritingToolsTextSuggestionData>(marker.data());
+        if (oldData.decoration != DocumentMarker::WritingToolsTextSuggestionData::Decoration::None)
+            return false;
+
+        auto offsetRange = OffsetRange { marker.startOffset(), marker.endOffset() };
+
+        markers.removeMarkers(node, offsetRange, { DocumentMarker::Type::WritingToolsTextSuggestion });
```

Comment:
Removing markers while iterating over them is concerning?

Reply:
I thought it was ok as long as they’re being added back in the same iteration, with the same ranges? Or is that still bad?